### PR TITLE
Op555 invalid home url generation

### PR DESCRIPTION
**Pull request by javitonino**
Fixes wrong redirection when:
- Using domainless urls
- Accessing by IP address
- Not logged in

In these circumstances, a URL of the form `http://cartodb.lan/user//login` was generated, with a blank username in the URL, which gives a 404 (no user found). This was fixed for domains in #6715 but not for the IP route.

### CODE REVIEW

**Comment by javitonino** [comment]
CR @juanignaciosl

File: config/initializers/carto_db.rb [hunk 1]
```diff
@@ -163,20 +163,25 @@ def self.subdomain_based_base_url(subdomain, org_username=nil, protocol_override
...
+    end
+  end
+
+  def self.domainless_base_domain(protocol_override = nil)
     protocol = self.protocol(protocol_override)
-    port = protocol == 'http' ? self.http_port : self.https_port
+    port = protocol == 'http' ? http_port : https_port
     if ip?(request_host)
-      "#{protocol}://#{request_host}#{port}/user/#{subdomain}"
+      "#{protocol}://#{request_host}#{port}"
```
Comment (juanignaciosl) [inline config/initializers/carto_db.rb]: Just checking: it was always adding the `/user/#{subdmain}` suffix, isn't it?

**Comment by juanignaciosl** [comment]
LGTM, but please test this in different scenarios (subdomainless in local and normal at staging, mainly).

Nice test addition btw :-)

File: config/initializers/carto_db.rb (hunk 1 above)
Comment (javitonino) [inline config/initializers/carto_db.rb]: Yes, even when `subdomain` was empty.

**Pull request merged**